core/src/test/java/org/apache/spark/memory/TaskMemoryManagerSuite.java

```diff
@@ -25,27 +26,37 @@
 
 public class TaskMemoryManagerSuite {
 
+  static TaskMemoryManager manager;
+
+  @After
+  public void after() {
+    Assert.assertEquals(0, manager.getMemoryConsumptionForThisTask());
```

Comment:
Add `Assert.assertNotNull(manager)`.

Also, probably minor, but if a test fails, these checks will probably fail in weird ways and might mask the original error... I'd try to insert a failure in a test and seeing what happens.